src/walletd/runtime.rs

```diff
@@ -1533,6 +1545,30 @@ impl Runtime {
                 self.clean_up_after_swap(&swap_id);
             }
 
+            Request::CheckpointMultipartChunk(checkpoint_multipart_chunk) => {
+                if let Some(checkpoint_request) = checkpoint_handle_multipart_receive(
+                    checkpoint_multipart_chunk,
+                    &mut self.pending_checkpoint_chunks,
+                )? {
+                    self.handle_rpc_ctl(endpoints, source, checkpoint_request)?;
+                }
+            }
+
+            Request::Checkpoint(request::Checkpoint { swap_id, state }) => match state {
+                CheckpointState::CheckpointWalletAlice(alice_state) => {
+                    info!("Restoring alice wallet for swap {}", swap_id);
+                    if !self.wallets.contains_key(&swap_id) {
+                        self.wallets.insert(swap_id, Wallet::Alice(alice_state));
+                    }
+                }
+                CheckpointState::CheckpointWalletBob(bob_state) => {
+                    info!("Restoring bob wallet for swap {}", swap_id);
+                    if !self.wallets.contains_key(&swap_id) {
+                        self.wallets.insert(swap_id, Wallet::Bob(bob_state));
+                    }
+                }
```

Comment:
maybe add the else branches for logging errors?

Reply:
Done in 5f87e31ff2fa217c207f0eff83778932eb3f0fd5